manual/src/refman/extensions/privatetypes.etex

```diff
@@ -45,7 +47,24 @@ end = struct
 end
 \end{caml_example*}
 Here, the @"private"@ declaration ensures that in any value of type
-"M.t", the argument to the "B" constructor is always a positive integer.
+"M.t", the argument to the "B" constructor is always a positive integer,
+because client of the "M" module must use the "b" function to construct
+"B" value.
+
+\begin{caml_example*}{verbatim}[error]
+(* this function compiles because it only uses pattern-matching *)
+let to_string = function
+  | M.A -> "M.A"
+  | M.B _ -> "M.B"
+
+(* this one does not because it constructs a type *)
+let make_b = M.B (-1)
+\end{caml_example*}
+
+Note that if the type is marked as private \emph{in the implementation},
+constructing a type is no longer possible within the implementation. It can
+be usefull when types are created externally (C extension) or used as
+marker in a GADT.
```

Comment:
I'm not fond of the word "marker" in this context, which as far as I understand is newly introduced terminology in the manual. If we want to write something about GADTs here, we should think carefully about the name we use for types used as type-level data in GADT indices.

Reply:
I propose documentation that includes some of this in #13994, which in particular mentions the use of private variants to index GADTs, and links to the private-types chapter of the manual. We could include a cross-reference to that new section here, but #13994 is not yet merged and will probably take more time than your own PR here, so I would suggest removing the mention of GADTs for now, and I can add this cross-reference as part of #13994 after your PR is merged.